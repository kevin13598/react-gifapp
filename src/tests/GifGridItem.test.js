import {shallow} from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';
import React from 'react';
import '@testing-library/jest-dom';

describe('Pruebas para el componente GifGridItem', () =>{

    const title = 'Un titulo';
    const url = 'https://localhosst/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('debe de mostrar el componente correctamente', ()=>{

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        //console.log(img.prop('src'));
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        /* const c = 'card animate__animated animate__fadeIn';
        expect(div.prop('className')).toBe(c); */

        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
    
    

});