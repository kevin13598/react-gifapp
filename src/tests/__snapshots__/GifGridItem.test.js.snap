// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhosst/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
